bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,325 @@
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We are proposing a way to package server and application inside a bootable jar. The jar contains everything needed to run the server.
+The packaged server can benefit from Galleon tooling and be trimmed down to user needs (thanks to Galleon layers).
+Such executable jar can then be run on bare metal and in cloud execution contexts.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13305[WFLY-13305]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1385[EAP7-1385]
+
+* https://issues.redhat.com/browse/WFCORE-4895[WFCORE-4895]
+
+* https://issues.redhat.com/browse/WFGP-169[WFGP-169]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly-maven-plugin[WildFly maven plugins]
+
+* https://github.com/wildfly/quickstart[WildFly Quickstarts]
+
+* https://start.microprofile.io/[Microprofile starter]
+
+
+== Requirements
+
+==== Bootable jar server vs Standalone server
+
+Although the bootable jar can be seen as a standalone server the following limitations apply:
+
+* CLI management operations that would imply a server restart are not supported.
+* :shutdown operation with restart (eg: by sending CLI operation :shutdown(restart=true)) is not supported. 
+  A warning is displayed and the bootable jar process is terminated.
+* Changes done to the management model (eg: using CLI or Web console) are not persisted between bootable jar executions.
+
+==== Defining a new WildFly Maven Plugin
+
+Existing maven plugins related to provision, build, and manage WildFly:
+
+* https://github.com/wildfly/galleon[org.jboss.galleon:galleon-maven-plugin]
+** Maven plugin goals to provision server are defined in the wildfly/galleon project
+* https://github.com/wildfly/galleon-plugins[org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin]
+** Maven plugin goals to build galleon feature-packs for WildFly are defined in wildfly/galleon-plugins project
+* https://github.com/wildfly/wildfly-maven-plugin[org.wildfly.plugins:wildfly-maven-plugin]
+** Maven plugin goals to manage a WildFly server and its deployments are defined in wildfly/wildfly-maven-plugin project
+
+It makes sense to define a new maven artifact for the plugin and not extend an existing plugin:
+
+* The new plugin depends on galleon-maven-plugin for the provisioning aspects but is WildFly specific. Galleon is only one aspect of it.
+* Adding goals to wildfly-maven-plugin would create confusion (start/run/shutdown are already existing goals that would need to be named differently for the bootable jar). The deployment model is also different.
+
+So the new plugin should be a new Maven artifact with the same groupID as the wildfly-maven-plugin: org.wildfly.plugins, artifactId: wildfly-jar-maven-plugin
```

Comment:
typo: "groupID" -> "groupId" and I'd replace the "," with "and" for better readability. Also a fullstop is missing here at the end of the sentence.